src/main/scala/com/fulcrumgenomics/fasta/SortSequenceDictionary.scala

```diff
@@ -0,0 +1,98 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.fasta
+
+import scala.collection.mutable.Builder
+import com.fulcrumgenomics.FgBioDef.{FgBioEnum, PathToSequenceDictionary}
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.CommonsDef._
+import com.fulcrumgenomics.commons.util.LazyLogging
+import com.fulcrumgenomics.fasta.SequenceMetadata.Keys
+import com.fulcrumgenomics.sopt._
+import com.fulcrumgenomics.util.Io
+import enumeratum.EnumEntry
+
+import scala.collection.immutable.IndexedSeq
+import scala.collection.mutable.ListBuffer
+
+
+@clp(description =
+  """
+    |Sorts a sequence dictionary file in the order of another sequence dictionary.
+    |
+    |The inputs are to two `*.dict` files.  One to be sorted, and the other to provide the order for the sorting.
+    |
+    |The output will be a "sequence dictionary", which is a valid SAM file, containing the version header line and one
+    |line per contig.  The fields of the entries in this dictionary will be the same as in input, but in the order of
+    |sortDictionary.
+  """,
+  group = ClpGroups.Fasta)
+class SortSequenceDictionary
+(@arg(flag='i', doc="Input sequence dictionary file to be sorted.") val input: FilePath,
+ @arg(flag='d', doc="Input sequence dictionary file containing contigs in the desired sort order.") val sortDictionary: PathToSequenceDictionary,
+ @arg(flag='o', doc="Output sequence dictionary file.") val output: PathToSequenceDictionary,
+ @arg(doc="Append contigs that have no matching contig in the existing dictionary to the end of the output dictionary. If false, such contigs will not be output") val appendMissingContigs: Boolean = true,
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(input)
+  Io.assertReadable(sortDictionary)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+      val input_dict = SequenceDictionary(input)
+      val sort_order_dict = SequenceDictionary(sortDictionary)
+
+      val new_order: Builder[SequenceMetadata, IndexedSeq[SequenceMetadata]] = IndexedSeq.newBuilder[SequenceMetadata]
+      var new_index_counter: Int = 0
+
+      sort_order_dict.foreach { sort_meta => 
+          input_dict.get(sort_meta.name) match {
+              case Some(in_meta) => {
+                  new_order += in_meta.copy(index=new_index_counter)
+                  new_index_counter += 1
+              }
```

Comment:
Can get rid of `new_index_counter` (see comment below).  
```suggestion
              case Some(in_meta) => new_order += in_meta
```